qa/tasks/vstart_runner.py

```diff
@@ -405,10 +400,41 @@ def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
 
         return args
 
+    @staticmethod
+    def split_args(args):
+        # Since Python's shell simulation can only work when commands are
+        # provided as a list of argumensts...
+        if isinstance(args, str):
+            if '\n' in args:
+                args = args.split('\n')
+                args = [a.split() for a in args]
+            else:
+                args = args.split()
+        return args
+
     # Wrapper to keep the interface exactly same as that of
     # teuthology.remote.run.
     def run(self, **kwargs):
-        return self._do_run(**kwargs)
+        """
+        Prepare arguments and get them executed.
+
+        We might have multiple commands in kwargs['args'] as list within list
+        or as string with newlines.
+        """
+        cmds = type(self).split_args(kwargs['args'])
```

Comment:
when performing method resolution for `instance.func()`, in addition to regular functions, both the static methods and class methods are considered. see https://docs.python.org/3/library/functions.html#staticmethod

so no need to call `split_args()` this way, could just put
```py
cmds = self.split(kwargs['args']
```

Reply:
True, I've made the change. Thanks!